example/test/function_component_test.dart

```diff
@@ -183,6 +183,134 @@ UseRefTestComponent(Map props) {
   ]);
 }
 
+final FancyInput = react.forwardRef((props, ref) {
+  final inputRef = useRef();
+
+  useImperativeHandle(
+    ref,
+    () {
+      print('FancyInput: useImperativeHandle re-assigns ref.current');
+      return {'focus': () => inputRef.current.focus()};
+    },
+
+    /// Because the return value of createHandle never changes, it is not necessary for ref.current
+    /// to be re-set on each render so this dependency list is empty.
+    [],
+  );
+
+  return react.input({
+    'ref': inputRef,
+    'value': props['value'],
+    'onChange': (e) => props['update'](e.target.value),
+    'placeholder': props['placeholder'],
+    'style': {'borderColor': props['hasError'] ? 'crimson' : '#999'},
+  });
+});
+
+final useImperativeHandleTestFunctionComponent =
+    react.registerFunctionComponent(UseImperativeHandleTestComponent, displayName: 'useImperativeHandleTest');
+
+UseImperativeHandleTestComponent(Map props) {
+  final city = useState('');
+  final state = useState('');
+  final error = useState('');
+  final message = useState('');
+
+  Ref cityEl = useRef();
```

Comment:
I think the `-El` suffix misleading here, since it's not actually an element, but rather a Map. Can we add typing to these as well to further show that? 

Maybe something like:
```dart
final cityRef = useRef<Map>();
```